packages/catalog-platformstate-writer/src/consumerServiceV2.ts

```diff
@@ -1,12 +1,202 @@
+          await updateDescriptorStateInPlatformStatesEntry(
+            dynamoDBClient,
+            primaryKeyCurrent,
+            descriptorStateToClientState(descriptor.state),
+            msg.version
+          );
+
+          // token-generation-states
+          const eserviceId_descriptorId = makeGSIPKEServiceIdDescriptorId({
+            eserviceId: eservice.id,
+            descriptorId: descriptor.id,
+          });
+          await updateDescriptorStateInTokenGenerationStatesTable(
+            eserviceId_descriptorId,
+            descriptor.state,
+            dynamoDBClient
+          );
+        } else {
+          const catalogEntry: PlatformStatesCatalogEntry = {
+            PK: primaryKeyCurrent,
+            state: descriptorStateToClientState(descriptor.state),
+            descriptorAudience: descriptor.audience[0],
+            version: msg.version,
+            updatedAt: new Date().toISOString(),
+          };
+
+          await writeCatalogEntry(catalogEntry, dynamoDBClient);
+
+          // token-generation-states
+          const eserviceId_descriptorId = makeGSIPKEServiceIdDescriptorId({
+            eserviceId: eservice.id,
+            descriptorId: descriptor.id,
+          });
+          await updateDescriptorStateInTokenGenerationStatesTable(
+            eserviceId_descriptorId,
+            descriptor.state,
+            dynamoDBClient
+          );
+        }
+      }
+
+      // flow for previous descriptor
+
+      if (
+        !previousDescriptor ||
+        previousDescriptor.state !== descriptorState.archived
+      ) {
+        return Promise.resolve();
+      } else {
+        const primaryKeyPrevious = makePlatformStatesEServiceDescriptorPK({
+          eserviceId: eservice.id,
+          descriptorId: descriptor.id,
+        });
+
+        await deleteCatalogEntry(primaryKeyPrevious, dynamoDBClient);
+
+        // token-generation-states
+        const eserviceId_descriptorId_previous =
+          makeGSIPKEServiceIdDescriptorId({
+            eserviceId: eservice.id,
+            descriptorId: previousDescriptor.id,
+          });
+        await updateDescriptorStateInTokenGenerationStatesTable(
+          eserviceId_descriptorId_previous,
+          descriptor.state,
+          dynamoDBClient
+        );
+      }
+    })
+    .with(
+      { type: "EServiceDescriptorActivated" },
+      { type: "EServiceDescriptorSuspended" },
+      async (msg) => {
+        const { eservice, descriptor } = parseEServiceAndDescriptor(
+          msg.data.eservice,
+          unsafeBrandId(msg.data.descriptorId)
+        );
+        const primaryKey = makePlatformStatesEServiceDescriptorPK({
+          eserviceId: eservice.id,
+          descriptorId: descriptor.id,
+        });
+        const catalogEntry = await readCatalogEntry(primaryKey, dynamoDBClient);
+
+        if (!catalogEntry) {
+          throw genericInternalError(
+            `Unable to find catalog entry with PK ${primaryKey}`
+          );
```

Comment:
this throw prevents partial reprocessing for this domain.
If the record is not found, let's return and skip the current event, I'll update the documentation accordingly

Reply:
Updated